projects/plugins/jetpack/extensions/blocks/premium-content/_inc/remove-block-keep-content.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * WordPress dependencies
+ */
+import { PluginBlockSettingsMenuItem } from '@wordpress/edit-post';
+import { select, dispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { transformToCoreGroup } from './transform-to-core-group';
+import { name } from '../index';
+
+function replaceBlockAndKeepContent() {
+	const block = select( 'core/editor' ).getSelectedBlock();
+	dispatch( 'core/block-editor' ).replaceBlock(
+		block.clientId,
+		transformToCoreGroup( block.innerBlocks )
```

Comment:
What do you think of not wrapping the inner blocks into a Group, but instead just spreading them into the content?
`replaceBlock` accepts an array of replacement blocks as well, so it might be as easy as:
```js
// untested code
replaceBlock( block.clientId, block.innerBlocks );
```

As the block is right now, I don't see an actual need for the Group, which might just end up complicating the UX.
Although playing devil's advocate, I guess that if we eventually add style options to the Premium Content's views, we would then need the Group in order to keep them after the removal. 🤔 

Then again, it might not be very useful? The PC views are intentionally very light, and encourage doing any stylings directly in the inner blocks.

I'm probably overthinking, but I'd love your opinion! 😄 

---

To clarify: this proposal would only affect when using the "remove and keep content" menu item; the transform to Group would obviously keep wrapping into a Group. 🙂

Reply:
I've initially thought of this while testing. My main argument against your approach would be that if we extract the premium content and don't move it to a group block, we could not re-convert it to premium content again, or would require some work on the user's side.

Reply:
> if we extract the premium content and don't move it to a group block, we could not re-convert it to premium content again, or would require some work on the user's side.

It's far from obvious but if you shift and click you can select multiple blocks, then choose transform to premium-content and it will wrap all selected blocks in a single premium content block.

Reply:
> It's far from obvious but if you shift and click you can select multiple blocks, then choose transform to premium-content and it will wrap all selected blocks in a single premium content block.

That's what I meant by "would require some work on the user's side" 😄

Reply:
I like the point that with a Group it would be trivial to transform back to PC, so let's keep the approach proposed here! 👍 

> That's what I've also observed, but it's actually a result of the group to PC transform, which is unrelated to this. I was planning to discuss it with the team before opening an issue on GH.

Feel free to open it to kickstart a conversation. ✨

Reply:
Actually, I just realised where the problem comes from and pushed a change that fixes it :D